### PR TITLE
146 altering the PhEval result types defined in the post-processing methods

### PR DESCRIPTION
**Pull request by yaseminbridges**
Altered the PhEval result types defined as dataclasses in the post-processing methods - this mainly involved changing the inheritance of the dataclasses to either the base parent class or the 'unranked' version of the result type. This involved refactoring how variables were obtained in the analysis results and how the results were parsed back into a dataclass (avoiding the use of dictionary keys)

### CODE REVIEW

File: src/pheval/post_processing/post_processing.py [hunk 1]
```diff
@@ -156,20 +162,18 @@ def _return_sort_order(sort_order_str: str) -> SortOrder:
...
     return _rank_pheval_result(sorted_pheval_result, sort_order)
 
 
 def _write_pheval_gene_result(
-    ranked_pheval_result: [RankedPhEvalGeneResult], output_dir: Path, tool_result_path: Path
+    ranked_pheval_result: [PhEvalResult], output_dir: Path, tool_result_path: Path
 ) -> None:
     """Write ranked PhEval gene result to tsv."""
-    ranked_result = pd.DataFrame([x.as_dict() for x in ranked_pheval_result])
+    ranked_result = pd.DataFrame([data.__dict__ for data in ranked_pheval_result])
```
Comment (matentzn) [inline src/pheval/post_processing/post_processing.py:176]: Why this change?

**Review by matentzn — APPROVED** [review]
If this works as expected this is a great simplification of the code! Nice.

File: src/pheval/post_processing/post_processing.py (hunk 1 above)
Comment (yaseminbridges) [reply src/pheval/post_processing/post_processing.py:176]: I ended up removing the method `as_dict()` method in the class as Jules wanted me to avoid creating those dictionary/dictionary keys manually

**Review by souzadevinicius — APPROVED** [review]
LGTM

**Review by julesjacobsen — APPROVED** [review]
Looks way more succinct. You could probably wrap the results in a ranked result and just delegate the calls from RankedResult.chr (for example) to Result.chr  by defining RankedResult as an interface implemented by RankedGeneResult and RankedVariantResult, then you can just call the standard constructor, like so RankedGeneResult(GeneResult, rank) and not need a static constructor to copy the input objects fields.
Still, this works and is better than the previous version.
